src/resolve_robotics_uri_py/resolve_robotics_uri_py.py

```diff
@@ -2,80 +2,151 @@
-    # This list specify the origin of each env variable:
-    # * GAZEBO_MODEL_PATH: Used in Gazebo Classic
-    # * ROS_PACKAGE_PATH: Used in ROS1
-    # * AMENT_PREFIX_PATH: Used in ROS2
-    # * SDF_PATH: Used in sdformat
-    # * IGN_GAZEBO_RESOURCE_PATH: Used in Ignition Gazebo <= 7
-    # * GZ_SIM_RESOURCE_PATH: Used in Gazebo Sim >= 7
-    env_list = ["GAZEBO_MODEL_PATH", "ROS_PACKAGE_PATH", "AMENT_PREFIX_PATH", "SDF_PATH", "IGN_GAZEBO_RESOURCE_PATH", "GZ_SIM_RESOURCE_PATH"]
-
-    # Preliminary step: if there is no scheme, we just consider this a path and we return it as it is
+    """
+    Resolve a robotics URI to an absolute filename.
+
+    Args:
+        uri: The URI to resolve.
+
+    Returns:
+        The absolute filename corresponding to the URI.
+
+    Raises:
+        FileNotFoundError: If no file corresponding to the URI is found.
+    """
+
+    # If the URI has no scheme, use by default the file://
     if "://" not in uri:
-        return pathlib.Path(uri)
+        uri = f"file://{uri}"
 
     # Get scheme from URI
     from urllib.parse import urlparse
+
+    # Parse the URI
     parsed_uri = urlparse(uri)
 
-    # We only support at the moment:
-    # file:// scheme: to pass a file path directly
-    # package:// : ROS-style package URI
-    # model:// : SDF-style model URI
-    if parsed_uri.scheme not in ["file", "package", "model"]:
-        raise FileNotFoundError(f"Passed URI \"{uri}\" use non-supported scheme {parsed_uri.scheme}")
+    # We only support the following URI schemes at the moment:
+    #
+    # * file://:    to pass an absolute file path directly
+    # * model://:   SDF-style model URI
+    # * package://: ROS-style package URI
+    #
+    if parsed_uri.scheme not in SupportedSchemes:
+        msg = "resolve-robotics-uri-py: Passed URI '{}' use non-supported scheme '{}'"
+        raise FileNotFoundError(msg.format(uri, parsed_uri.scheme))
 
-    model_filenames = []
+    # Strip the URI scheme
+    uri_path = uri.replace(f"{parsed_uri.scheme}://", "")
 
     if parsed_uri.scheme == "file":
-        model_filenames.append(uri.replace("file:/", ""))
-
-    if parsed_uri.scheme == "package" or parsed_uri.scheme == "model":
-        uri_path = uri.replace(f"{parsed_uri.scheme}://","")
-        for folder in get_search_paths_from_envs(env_list):
-            candidate_file_name = folder / pathlib.Path(uri_path)
-            if (candidate_file_name.is_file()):
-                if candidate_file_name not in model_filenames:
-                    model_filenames.append(candidate_file_name)
-
-    if model_filenames:
-        if (len(model_filenames) > 1):
-            warnings.warn(f"resolve-robotics-uri-py: Multiple files ({pathlist_list_to_string(model_filenames)}) found for uri \"{uri}\", returning the first one.")
+        # Ensure the URI path is absolute
+        uri_path = uri_path if uri_path.startswith("/") else f"/{uri_path}"

```

Comment:
This logic is wrong on Windows, and probably the reason why the tests are failing on Windows. Can't we just pass `uri_path` to `pathlib.Path` ? If it is not an absolute path, an error will be raised, as it should be.

Reply:
Ah I see, the problem is that you removed the special handling in `        model_filenames.append(uri.replace("file:/", ""))` for the `file:/`. Any specific reason to do that?

Reply:
Let's consider an URI in the form `file://absolute/path/to/file.txt`, and the URI without scheme `/absolute/path/to/file.txt`. With this PR, URIs without schemes fall back to `file://`.

When the scheme is stripped, the path becomes either `absolute/path/to/file.tx` or remains `/absolute/path/to/file.txt`. At least in Ubuntu, I need to add a trailing `/`.

Before this PR, the logic was always removing `file:/` (note the missing trailing `/`). Leaving an extra `/` in case the path was already absolute. I'll make some test in CI on windows to figure out what's wrong there.

Reply:
> When the scheme is stripped, the path becomes either absolute/path/to/file.tx or remains /absolute/path/to/file.txt. At least in Ubuntu, I need to add a trailing /.

I am a bit confused. The scheme is `file:/`, so if you remove it you remain with `/absolute/path/to/file.txt`, in which context you will have `absolute/path/to/file.txt` after stripping `file:/` ?

Reply:
I never strip `file:/`. The new logic always strips `file://` and adds a leading `/` to the resulting path if it doesn't start already with `/`.

Reply:
Yes, my question is exactly: why can't we strip `file:/` as before the PR?

Reply:
Actually, I think that the logic (and the test) were at least incomplete also before. It seems that before it was kind of assuming that file uri in Windows were `file:/C:/etc/etc`, but according to https://datatracker.ietf.org/doc/rfc8089/ also `file:C:/etc/etc` and `file:///C:/etc/etc` are supported, and actually the latter is the one used in https://en.wikipedia.org/wiki/File_URI_scheme . However, for sure it seems that here instead now `file://C:/etc/etc` is used?

Reply:
Good to know, I think we're not yet close to support the full `file://` specification. Maybe we can try to be 100% compatible in a new PR.

Reply:
To be honest, I am not even sure we want to be, whatever subset works fine for the average robotics users for us is ok.

Reply:
However, if I am not wrong in https://github.com/ami-iit/resolve-robotics-uri-py/pull/11/files#diff-1e7414883e02e7f25fddd798742a4cd9cc87486af18923f0efb6940c30f49ebeR95 we are building uris like `file://C:/path/to`, that from what I understand are not compatible with the spec.

Reply:
Also in https://github.com/ami-iit/resolve-robotics-uri-py/pull/11/files#diff-bde2b03789a37955655850a5b32d09d6775334952d2f4fbfe5510b5e005223ffR135 it seems that we are building `file://C:/path/to` URIs.